larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.cc

```diff
@@ -0,0 +1,129 @@
+/**
+ *  @file   larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.cc
+ *
+ *  @brief  Implementation of the discrete probability helper helper class.
+ *
+ *  $Log: $
+ */
+
+#include "larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.h"
+
+namespace lar_content
+{
+
+template <typename T>
+float LArDiscreteProbabilityHelper::CalculateCorrelationCoefficientPValueFromPermutationTest(const T &t1, const T &t2, 
+    std::mt19937 &randomNumberGenerator, const size_t nPermutations)
```

Comment:
I opted for size_t based on internet hive mind suggestions e.g.: https://www.viva64.com/en/a/0050/
The rough TL;DR was use size_t when you're describing the size of something e.g. size_t for a vector but unsigned int for a count
I do, however, concede that consistency is important.
It's easy enough for me to swap the code so I am happy for you to have the deciding vote on this @johnmarshall80

Reply:
Ah, I see that in the code you've pointed out, I really should be using unsigned int here as its a count....
I will change this.

Reply:
I've opted for a global swap.  Grepping did reveal a few classes not related to the PR that use size_t as well.  I left them be but it might be worth looking at them down the line.